api/v1alpha1/ndb_api_helpers.go

```diff
@@ -159,18 +159,68 @@ func GetNoneTimeMachineSLA(ctx context.Context, ndbclient *ndbclient.NDBClient)
 	return sla, fmt.Errorf("NONE TimeMachine not found")
 }
 
-// Fetches all the profiles and returns a map of OOB profiles
-// Returns an error if one or more OOB profiles is not found
-func GetOOBProfiles(ctx context.Context, ndbclient *ndbclient.NDBClient, dbType string) (profileMap map[string]ProfileResponse, err error) {
+// This method populates the profileMap with all the values for the profileType available
+func GetProfiles(ctx context.Context, ndbclient *ndbclient.NDBClient, dbType string, profiles Profiles) (profileMap map[string]ProfileResponse, err error) {
+
+	log := ctrllog.FromContext(ctx)
+
 	// Map of profile type to profiles
 	profileMap = make(map[string]ProfileResponse)
 
-	profiles, err := GetAllProfiles(ctx, ndbclient)
+	allProfiles, err := GetAllProfiles(ctx, ndbclient)
 
+	err = GetOOBProfiles(profileMap, allProfiles, dbType)
 	if err != nil {
 		return
 	}
 
+	log.Info("Received Input Profiles = ", "Received Input Profiles", profiles)
+
+	customProfileOptions := [...]string{PROFILE_TYPE_COMPUTE, PROFILE_TYPE_SOFTWARE, PROFILE_TYPE_NETWORK, PROFILE_TYPE_DATABASE_PARAMETER}
+	for _, profileValue := range customProfileOptions {
+		err = MatchAndSetProfile(ctx, profileValue, profiles.Compute, allProfiles, profileMap)
+		if err != nil {
+			return
+		}
+	}
+
+	return
+
+}
+
+/*
+This function does:
+(a) Id level matching with profiles in allProfiles
+(b) If Id level match is successful, flow proceeds to match based on versionId
+
+	When matched, the latestVersionId is overridden with the versionId as it is this attribute while dbProvisioning which is used forprofileType versionId
```

Comment:
Thanks for adding a detailed comment, helps in understanding the complexities of the logic.